packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.ts

```diff
@@ -1228,13 +1237,57 @@ export class AmplifyAuthCognitoStack extends cdk.Stack implements AmplifyAuthCog
       this.hostedUIDomainResource.addDependency(this.hostedUICustomResource);
     }
   };
-}
 
-type OAuthMetaData = {
-  AllowedOAuthFlows?: Array<string>;
-  AllowedOAuthFlowsUserPoolClient?: boolean;
-  AllowedOAuthScopes?: Array<string>;
-  CallbackURLs?: Array<string>;
-  LogoutURLs?: Array<string>;
-  SupportedIdentityProviders?: Array<string>;
-};
+  createHostedUIProvidersResources = (props: CognitoStackOptions) => {
+    if (!props.hostedUIProviderMeta) {
+      return;
+    }
+
+    const authCfnTemplatePath = getResourceCfnTemplatePath(getBackendDirPath(), 'auth', props.resourceName);
+    const authCfnTemplate: Template | undefined = readJson(authCfnTemplatePath, { throwIfNotExist: false });
```

Comment:
is `authCfnTemplate` an output of this logic?
if yes then is outcome idempotent ?

Reply:
> is authCfnTemplate an output of this logic?
if yes then is outcome idempotent ?


Discussed this offline and will add some unit tests to address this log.

Reply:
> Also one more thing , why are we reading build cfn file ?
Can we get this info from cli-inputs.json ?

The CLI inputs file doesn't change with this change, just the CFN template files